api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -160,7 +204,24 @@ object ServiceReader {
   }
 }
 
-case class SelfDescribingServiceCallStub[Id, Request, Response](method: Method, idType: Type, requestType: Type, responseType: Type) extends SelfDescribingServiceCall[Id, Request, Response] {
-  def invoke(id: Id, request: Request) = throw new NotImplementedError("This service call is a stub that should not be invoked")
-  def methodName = method.getName
-}
+/**
+ * Internal API.
+ *
+ * Service call holder that holds the original method reference that was passed to the descriptor when the service
+ * descriptor was defined.
+ *
+ * @param methodReference The method reference (a lambda object).
+ */
+private[lagom] case class MethodRefServiceCallHolder(methodReference: Any) extends ServiceCallHolder
+
+/**
+ * Internal API.
+ *
+ * Service call holder that's used by the service router and client implementor to essentially the raw id to and from
+ * invocations of the service call method.
+ */
+private[lagom] trait MethodServiceCallHolder extends ServiceCallHolder {
+  val method: Method
```

Comment:
Yes, the intent is that this trait should only ever be a wrapper around a method.  If there were ever a reason for an implementation to implement it using a `def`, then assumptions made elsewhere in the code may be violated.

Basically I want to ensure that if another type of service call holder is ever implemented that behaves differently, that a new type is introduce to handle that, and that all the places that use it around Lagom will be updated to handle that new type.  This existing type shouldn't be reused for another purpose.